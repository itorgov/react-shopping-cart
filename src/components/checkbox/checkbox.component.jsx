import React from 'react';
import classNames from "classnames";

import styles from './checkbox.module.scss';

export const Checkbox = ({children: label}) => (
    <label className={classNames([
        styles.checkbox,
        'my-4'
    ])}
    >
        <input type="checkbox"/>
        <span className={styles.checkmark}/>
        {label}
    </label>
);
