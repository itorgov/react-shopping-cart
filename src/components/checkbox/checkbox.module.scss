.checkbox {
  display: flex;
  align-items: center;
  position: relative;
  cursor: pointer;
  user-select: none;
  font-size: 1.3rem;

  & input {
    display: none;
  }

  & input:checked ~ .checkmark:after {
    display: block;
  }

  & .checkmark {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-right: 1rem;
    height: 2rem;
    width: 2rem;
    background-color: var(--color-white);

    &:after {
      content: "";
      display: none;
      width: 0.5rem;
      height: 1rem;
      border: solid var(--color-text-dark);
      border-width: 0 2px 2px 0;
      transform: rotate(45deg);
    }
  }
}
