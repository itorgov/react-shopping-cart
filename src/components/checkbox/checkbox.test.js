import React from "react";
import {shallow} from "enzyme";

import {Checkbox} from './checkbox.component';

describe('<Checkbox />', () => {
    it('expect to render', () => {
        expect(shallow(<Checkbox />)).toMatchSnapshot();
    });
});
