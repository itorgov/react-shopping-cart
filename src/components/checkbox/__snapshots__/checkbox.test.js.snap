// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> expect to render 1`] = `
<label
  className="checkbox my-4"
>
  <input
    type="checkbox"
  />
  <span
    className="checkmark"
  />
</label>
`;
